tests/test_api.py

```diff
@@ -125,3 +125,54 @@ def test_record_performance_file_name(self):
         perf_name = __file__.replace('.py', '.file_name.perf.yml')
         with self.record_performance(file_name=perf_name):
             caches['default'].get('foo')
+
+
+class FileNameTests(TestCase):
+
+    default_perf_name = __file__.replace('.py', '.perf.yml')
+    custom_perf_name = 'custom.perf.yml'
+    custom_path_perf_name = os.path.join(
+        os.path.dirname(__file__),
+        'performance_tests',
+        os.path.basename(__file__).replace('.py', '.perf.yml')
+    )
+
+    def setUp(self):
+        super(FileNameTests, self).setUp()
+        self.ensure_no_file()
+
+    def tearDown(self):
+        self.ensure_no_file()
+        super(FileNameTests, self).tearDown()
+
+    def ensure_no_file(self):
+        for name in (self.custom_perf_name, self.custom_path_perf_name):
+            try:
+                os.unlink(name)
+            except OSError as exc:
+                if exc.errno != errno.ENOENT:
+                    raise
+
+        try:
+            os.rmdir(os.path.dirname(self.custom_path_perf_name))
+        except OSError as exc:
+            if exc.errno != errno.ENOENT:
+                raise
+
+    def test_default_filename(self):
+        with record():
+            caches['default'].get('foo')
+
+        assert(os.path.exists(self.default_perf_name))
+
+    def test_custom_filename(self):
```

Comment:
This test was basically the same as `RecordBlankFileTests.test_one` apart from checking the file exists, so I merged them. Also `RecordBlankFileTests` etc. needed updating to use `path` instead of `file_name` because our tests shouldn't be using the deprecated arguments 😄